### PR TITLE
add documentation

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/giantswarm/issues/18077

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,29 +2,74 @@
...
-8. Change the badge (with style=shield):
-   https://circleci.com/gh/giantswarm/upgrade-schedule-operator.svg?style=shield&circle-token=TOKEN_FOR_PRIVATE_REPO
-   If this is a private repository token with scope `status` will be needed.
-
-[generate]: https://github.com/giantswarm/upgrade-schedule-operator/generate
+This operator is intended to automate scheduling, triggering and notification for upgrades of giant swarm workload clusters.
+
+## prerequisites
+
+Running this operator alone will enable the automatic scheduling and triggering of upgrades on a management cluster in any provider.
```
Comment (AverageMarcus) [inline README.md:9]: `upgrades on a management cluster`

Is that correct? It's not upgrading the management cluster is it? Or is this referring to triggering a job that runs on the management cluster to perform the upgrade?

File: README.md [hunk 2]
```diff
@@ -2,29 +2,74 @@
...
+
+To schedule your upgrade, simply add the following annotations to the `Cluster` CR to specify the time and desired version.
+```
+annotations:
+  alpha.giantswarm.io/update-schedule-target-release: 15.2.1
+  alpha.giantswarm.io/update-schedule-target-time: "15 Sep 21 08:00 UTC"
+```
+Please note that the release version has to be an existing release higher than the current release version.
+The time has to be given in RFC822 format and UTC.
+Furthermore, only times that are at least 16 minutes in the future but not more than 6 months are accepted.
```
Comment (AverageMarcus) [inline README.md:29]: Just curious - where's the 16 minutes come from?

File: README.md [hunk 3]
```diff
@@ -2,29 +2,74 @@
...
+
+## what happens next
+
+For your scheduled upgrades you should be able to see the remaining time in the logs of the `upgrade-schedule-operator`
+
+```
+2021-09-14T17:23:29.605Z	INFO	controllers.Cluster	The scheduled update time is not reached yet. Cluster will be upgraded in 14h37m0s at 2021-09-15 08:00:00 +0000 UTC.	{"cluster": "default/xyz01"}
+```
+Around 10-15 minutes before the scheduled upgrade, a slack message should appear in the specified slack channel.
```
Comment (AverageMarcus) [inline README.md:38]: Where is the slack channel specified? Can this vary by cluster?

**Review by AverageMarcus — APPROVED** [review]
Looks good to me. My comments are more just for my understanding / limited knowledge currently.

File: README.md (hunk 1 above)
Comment (anvddriesch) [reply README.md:9]: Uh yeah maybe the wording here is back. basically what I mean is, if you add the controller to your MC you can upgrade the WCs on it. I'll try to word it better.

File: README.md (hunk 2 above)
Comment (anvddriesch) [reply README.md:29]: We send a notice about the upgrade ~15 minutes before so that everyone can acknowledge it's going to happen.

Comment (AverageMarcus) [reply README.md:29]: I thought that might be the case but just wanted to check :)

File: README.md (hunk 3 above)
Comment (anvddriesch) [reply README.md:38]: I thought there was some doc in the `event-exporter-app` repo but looks like there isn't. I believe you need to create a token for a channel to the `Giant Swarm Cluster Upgrade` app in slack and then add it to the config for the MC: https://github.com/giantswarm/config
I think how it's set up at the moment, we can have one per MC but sometimes they are shared for the different MCs of the same customer.

**Review by njuettner — APPROVED** [review]
lgtm

**Pull request merged**